### PR TITLE
Feature/fix attributes

### PR DESCRIPTION
**Pull request by oezelenes**

### CODE REVIEW

File: src/main/java/de/retest/web/WebElementPeer.java [hunk 1]
```diff
@@ -89,10 +90,9 @@ public Integer extractSuffix() {
 
 	protected MutableAttributes retrieveStateAttributes() {
 		final MutableAttributes state = new MutableAttributes();
-
 		for ( final String key : webData.getKeys() ) {
 			final String attributeValue = webData.getAsString( key );
-			if ( attributeValue != null ) {
+			if ( attributeValue != null && !isIdentifyingAttributesKey( key ) ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/WebElementPeer.java]: Does that mean we exclude identifying attributes from the actual state?

File: src/main/java/de/retest/web/AttributesConfig.java [hunk 2]
```diff
@@ -17,6 +19,13 @@
 	public static final String WIDTH = "width";
 	public static final String HEIGHT = "height";
 
+	public static final String TAG_NAME = "tagName";
+
+	public static final String TEXT = "text";
+
+	private static final HashSet<String> identifyingAttributeKeys = new HashSet<>( Arrays.asList( ABSOLUTE_HEIGHT,
```
Comment (beatngu13) [inline src/main/java/de/retest/web/AttributesConfig.java]: Left-hand side should be `Set`.

File: src/main/java/de/retest/web/AttributesConfig.java [hunk 3]
```diff
@@ -35,4 +44,8 @@ public AttributesConfig( @JsonProperty( "attributes" ) final List<String> attrib
 		return identifyingAttributes;
 	}
 
+	static boolean isIdentifyingAttributesKey( final String key ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/AttributesConfig.java]:
```suggestion
	static boolean isIdentifyingAttribute( final String key ) {
```

File: src/main/java/de/retest/web/AttributesConfig.java (hunk 2 above)
Comment (beatngu13) [reply src/main/java/de/retest/web/AttributesConfig.java]: Plus: it's somewhat odd to have two identifying attribute sets/lists. The problem is actually the underlying YAML, we should probably change.

File: src/main/java/de/retest/web/AttributesConfig.java [hunk 4]
```diff
@@ -17,6 +19,13 @@
 	public static final String WIDTH = "width";
 	public static final String HEIGHT = "height";
 
+	public static final String TAG_NAME = "tagName";
```
Comment (beatngu13) [inline src/main/java/de/retest/web/AttributesConfig.java]: Isn't tag name = type?

File: src/main/java/de/retest/web/WebElementPeer.java (hunk 1 above)
Comment (oezelenes) [reply src/main/java/de/retest/web/WebElementPeer.java]: No, the attribute name "state" is a bit misleading there I guess. The method is used to retrieve the attributes for an element. So we just exclude already existing IdentifyingAttributes content.

File: src/main/java/de/retest/web/AttributesConfig.java (hunk 4 above)
Comment (oezelenes) [reply src/main/java/de/retest/web/AttributesConfig.java]: We call it type but actaully use tagName to extract it from the data.

**Comment by beatngu13** [comment]
Did some refactoring trying to make clear what we do with `attributes.yaml` and how it is used in `getAllElementsByPath.js`. We can probably also use `AttributesUtil` to bundle filter methods and stuff. What do you think @oezelenes @modulo11?

**Pull request merged**